### PR TITLE
i18n(ar): Translate Astro-Components page.

### PR DESCRIPTION
**Pull request by Somespi**
#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

- What does this PR change? Give us a brief description.

translating `core-concepts/astro-components` page to Arabic, although I'm not sure of some words, so I used some docs as a reference.

### CODE REVIEW

**Comment by yhyadev** [comment]
You forgot to change the navbar, It is still saying "Components" instead of "المكونات".

**Comment by Somespi** [comment]
> You forgot to change the navbar, It is still saying "Components" instead of "المكونات".

sure its, thanks for notifying me, I fixed it.

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+:::
+
+### قالب المكوّن
+
+قالب المكوّن موجود أسفل سياج الكود "code fence" و يحدد إخراج HTML الخاص بمكوّنك
+
+إذا قمت بكتابة تنسيق HTML العادي هنا، فسيقوم المكوّن الخاص بك بتصيير ذلك الـ HTML في أي صفحة أسترو يتم إستيرادها و إستخدامها 
+
+و مع ذلك، يدعم [ بناء قالب المكوّن لأسترو](/ar/core-concepts/astro-syntax/) كذلك **تعابير جافاسكربت**، وسوم [`<style>`](/ar/guides/styling/#styling-in-astro) و [`<script>`](/guides/client-side-scripts/#using-script-in-astro) لأسترو و المكوّنات المستوردة و [توجيهات أسترو الخاصة](/reference/directives-reference/). البيانات و القيم المعرّفة في البرنامج النصي للمكوّن يمكن إستخدامها في قالب المكوّن لإنتاج HTML  مُنشأ ديناميكيًا.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:67]: - "ومع ذلك، يدعم هيكل قالب المكوّن في أسترو تعابير جافاسكربت .." بدلًا من المكتوب.
- حذف المسافات بعد واو العطف
- "استخدامها" همزة وصل

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+  {myFavoritePokemon.map((data) => <li>{data.name}</li>)}
+</ul>
+
+<!-- Use a template directive to build class names from multiple strings or even objects! -->
+<p class:list={["add", "dynamic", {classNames: true}]} />
+```
+
+## التصميم القائم على المكوّنات
+
+المكوّنات مصممة لتكون قابلة لإعادة الاستخدام وقابلة للتركيب. يمكنك استخدام المكوّنات داخل مكوّنات أخرى لبناء واجهات مستخدم متقدمة بشكل أكبر وأكثر تطوراً. على سبيل المثال، يمكن استخدام مكوّن "زر" (Button) لإنشاء مكوّن "مجموعة أزرار" (ButtonGroup):
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:100]:
```suggestion
المكوّنات مصممة لتكون قابلة **لإعادة الاستخدام** و**التركيب**. يمكنك استخدام مكوّنات داخل مكوّنات أخرى لبناء واجهات مستخدم متقدمة بشكل أكبر وأكثر تطورًا. على سبيل المثال، يمكن استخدام مكوّن "زر" (Button) لإنشاء مكوّن "مجموعة أزرار" (ButtonGroup):
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+  <Button title="Button 1" />
+  <Button title="Button 2" />
+  <Button title="Button 3" />
+</div>
+```
+
+
+## خاصيّات المكوّن
+
+يمكن لمكوّن أسترو أن يعرّف و يستقبل الخاصيّات "props". هذه الخاصيّات "props" تصبح متاحة بعدها لقالب المكوّن لتصيير HTML. الخاصيّات "props" متاحة في `Astro.props` .بشكل عام في البرنامج النصّي  للـfrontmatter  الخاص بك
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:117]:
```suggestion
يمكن لمكوّن أسترو تعريف الخاصيات وقبولها "props". هذه الخاصيّات تصبح متاحة لقالب المكوّن لتصيير HTML. الخاصيّات متاحة على `Astro.props` بشكل عام في  الـfrontmatter الخاص بك.
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+هنا مثال لمكوّن يستقبل خاصيّة `greeting` و خاصيّة `name`. لاحظ أن الخاصيّات "props" التي سيتم استقبالها مفككة من object `Astro.props` العام.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:119]:
```suggestion
هنا مثال لمكوّن يستقبل خاصيّة `greeting` وخاصيّة `name`. لاحظ أن الخاصيّات التي سيتم استقبالها مفككة من object `Astro.props` العام.
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+```astro "Astro.props"
+---
+// src/components/GreetingHeadline.astro
+// Usage: <GreetingHeadline greeting="Howdy" name="Partner" />
+const { greeting, name } = Astro.props;
+---
+<h2>{greeting}, {name}!</h2>
+```
+هذا المكوّن، عندما يُستورد و يُصيّر في مكوّنات أسترو اخرى، مخططات أو صفحات، يمكن أن يتم تمرير هذه الخاصيّات "props" كسمات "attributes":
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:129]:
```suggestion
عند استيراد هذا المكوّن وتصييره في مكونات أو تخطيطات أو صفحات أسترو أخرى، يمكن تمرير هذه الخاصيّات "props" كسمات "attributes":

```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+// src/components/GreetingCard.astro
+import GreetingHeadline from './GreetingHeadline.astro';
+const name = "Astro"
+---
+<h1>Greeting Card</h1>
+<GreetingHeadline greeting="Hi" name={name} />
+<p>I hope you have a wonderful day!</p>
+```
+
+بالإمكان ايضًا تعريف الخاصيّات "props" الخاصة بك مع TypeScript باستخدام واجهة `Props`. سيقوم أسترو تلقائيًا باكتشاف واجهة `Props` في معلوماتك الأوّليّة "frontmatter"  و يعطي تحذيرات أو أخطاء. يمكن أيضًا تحديد القيم الإفتراضية لهذه الخاصيّات "props" عند استخراجها من `Astro.props`.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:142]:
```suggestion
بالإمكان أيضًا تعريف الخاصيّات الخاصة بك مع TypeScript باستخدام واجهة `Props`. سيقوم أسترو تلقائيًا باكتشاف واجهة `Props` في الـ "frontmatter" الخاص بك وإعطاء تحذيرات أو أخطاء إن وجد. يمكن أيضًا إعطاء هذه الخاصيات قيمًا افتراضية عند استخراجها من `Astro.props`.
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+interface Props {
+  name: string;
+  greeting?: string;
+}
+
+const { greeting = "Hello", name } = Astro.props;
+---
+<h2>{greeting}, {name}!</h2>
+```
+يمكن إعطاء خاصيّات "props" المكوّنات قيم إفتراضية عندما لا يتم توفير أي منها.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:156]:
```suggestion
يمكن إعطاء خاصيّات المكوّنات قيمًا افتراضية عندما لا يتم توفير أي منها.
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+---
+// src/components/GreetingHeadline.astro
+const { greeting = "Hello", name = "Astronaut" } = Astro.props;
+---
+<h2>{greeting}, {name}!</h2>
+```
+
+## المداخل
+
+`<slot />` هو عنصر نائب لمحتوى HTML الخارجي، مما يسمح لك بحقن (أو "ادخال") عناصر داخلية من ملفّات اخرى الى قالب مكوّنك.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:168]:
```suggestion
`<slot />` هو عنصر نائب لمحتوى HTML الخارجي، مما يسمح لك بحقن (أو "ادخال") عناصر من ملفّات أخرى الى قالب مكوّنك.
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+
+بشكل افتراضي، سيتم تصيير كل العناصر الداخلية الممرّرة لمكوّن في `<slot />` الخاص بها.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:170]: - using "الداخلية" does not make sense, use "التابعة if you must.
- الخاص به (مكوّن)، وليس بها.

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+// src/pages/fred.astro
+import Wrapper from '../components/Wrapper.astro';
+---
+<Wrapper title="Fred's Page">
+  <h2>All about Fred</h2>
+  <p>Here is some stuff about Fred.</p>
+</Wrapper>
+```
+
+هذ النمط هو أساس [مكوّنات تخطيط أسترو](/ar/core-concepts/layouts/): صفحة كامل من محتوى HTML يمكن أن "تُلف" بواسطة وسوم `<SomeLayoutComponent></SomeLayoutComponent>` و يتم ارسالها الى المكوّن ليتهم تصييرها داخل العناصر المشتركة المحددة هناك.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:205]: component is not plural, use "مكوّن تخطيط أسترو"

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+
+### المداخل المسماة
+
+يمكن كذلك لمكوّن أسترو أن يحتوي على مداخل مسماة "named slots". يسمح لك هذا بتمرير عناصر HTML فقط مع اسم المدخل المقابل الى موقع المدخل
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:209]: full stop

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+</Wrapper>
+```
+
+هذ النمط هو أساس [مكوّنات تخطيط أسترو](/ar/core-concepts/layouts/): صفحة كامل من محتوى HTML يمكن أن "تُلف" بواسطة وسوم `<SomeLayoutComponent></SomeLayoutComponent>` و يتم ارسالها الى المكوّن ليتهم تصييرها داخل العناصر المشتركة المحددة هناك.
+
+### المداخل المسماة
+
+يمكن كذلك لمكوّن أسترو أن يحتوي على مداخل مسماة "named slots". يسمح لك هذا بتمرير عناصر HTML فقط مع اسم المدخل المقابل الى موقع المدخل  
+
+يمكن تسمية الفتحات "slots" باستخدام سمة "attribute" `name`:
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:211]: you either use فتحات or مداخل to reduce confusion

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+  <slot name="after-header"/>  <!--  children with the `slot="after-header"` attribute will go here -->
+  <Logo />
+  <h1>{title}</h1>
+  <slot />  <!--  children without a `slot`, or with `slot="default"` attribute will go here -->
+  <Footer />
+  <slot name="after-footer"/>  <!--  children with the `slot="after-footer"` attribute will go here -->
+</div>
+```
+
+لإدخال محتوى HTML الى مدخل معين، يمكنك إستخدام سمة "attribute" `slot` على أي عنصر داخلي لتحديد اسم المدخل. سيتم إدخال بقية العناصر الداخلية الى المدخل الإفتراضي (غير مسمى) `<slot />`.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:233]: - "لحقن" بدل "لإدخال"
- ماله داعي كتابة ترجمة كلمة ما كل ما يتم استخدامها، مرة وحدة تكفي
- عنصر تابع أدق من عنصر داخلي

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+---
+<Wrapper title="Fred's Page">
+  <img src="https://my.photo/fred.jpg" slot="after-header">
+  <h2>All about Fred</h2>
+  <p>Here is some stuff about Fred.</p>
+  <p slot="after-footer">Copyright 2022</p>
+</Wrapper>
+```
+
+إستخدم سمة "attribute" `slot="my-slot"` على العنصر الداخلي الذي تريد تمريره الى عنصر نائب مطابق لـ `<slot name="my-slot" />` في مكوّنك.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:248]: - فعل الأمر ما تكتب همزته قطع، استخدم
- احذف التراجم الإنجليزية الغير ضرورية

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+
+لاحظ أن المدخلات المسماة "named slots" يجب أن تكون تابعة بشكل مباشر للمكوّن. لا يمكنك ان تمرر مدخلات مسماة داخل عناصر متداخلة.
+
+:::tip
+يمكن ايضًا أن تمرر المدخلات المسماة الى [مكوّنات أطر واجهة المستخدم](/ar/core-concepts/framework-components/)!
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:253]: أيضًا، إلى

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+:::
+
+
+### المحتوى الإحتياطي للمداخل
+
+بإمكان المداخل ايضًا ان تصّير **محتوى احتياطي**. عندما لا يكون هنالك عنصر داخلي مطابق مُرّرَ الى المدخل، عنصر `<slot />` سيُصيّر عنصر داخلي نائب خاص به.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:259]: "عند عدم وجود عناصر تابعة مُمررة إلى "الفتحة أو المدخل - اختر وحدة"، عنصر .."

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+  <h1>{title}</h1>
+  <slot>
+    <p>This is my fallback content, if there is no child passed into slot</p>
+  </slot>
+  <Footer />
+</div>
+```
+
+
+## مكوّنات HTML
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:282]: proofread the whole section
- inconsistent punctuation
- waslas
- src/pages/

**Review by g9wz — CHANGES_REQUESTED** [review]
left "some" comments 😅

great work otherwise

**Comment by Somespi** [comment]
I've addressed all the suggestions, and upon review, I noticed that I had overlooked several mistakes. Thank you for correcting me 😅

**Comment by g9wz** [comment]
> @g9wz I've addressed all the suggestions, and upon review, I noticed that I had overlooked several mistakes. Thank you for correcting me 😅

I know what it is like to deal with rtl in markdown, don't worry about it 😁

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+title: المكوّنات
+description: مقدمة إلى بناء المكوّنات في .astro 
+i18nReady: true
+---
+
+**مكوّنات أسترو** هي اللبنات الأساسية لأي مشروع أسترو. إنها مكوّنات قوالب HTML فقط بلا وقت تشغيل من جانب العميل. يمكنك التعرف على مكوّن ويب أسترو عن طريق إمتداد الملف: `.astro`
+
+مكونات أسترو مرنة للغاية. غالبًا ما يحتوي مكوّن أسترو على بعض **عناصر واجهة المستخدم القابلة لإعادة الاستخدام**، مثل العنوان أو بطاقة التعريف الشخصي. في أوقات أخرى، قد يحتوي مكون أسترو على أجزاء HTML متكررة، مثل مجموعة من علامات `<meta>` الشائعة في جعل العمل مع تحسين محركات البحث (SEO) أسهل. يمكن حتى لمكوّنات أسترو أن تحتوي على تخطيط صفحة كاملة.
+
+أهم شيء يجب معرفته عن مكوّنات أسترو هو أنها **لا تُصيّر على جانب العميل**. يحوَّل المكون إلى HTML إما في وقت البناء أو حسب الطلب باستخدام [ (SSR) العرض على الخادم](/ar/guides/server-side-rendering/) . يمكنك تضمين كود جافاسكربت داخل  "frontmatter"ّ  في مكوّن الويب الخاص بك. وسيتم إزالة كل ذلك من الصفحة النهائية التي يتم إرسالها إلى متصفحات مستخدميك. النتيجة هي موقع ويب أسرع، بدون أثر لجافاسكربت يضاف بشكل إفتراضي .
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:11]: inconsistent use of quotation marks around frontmatter

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+
+// Access passed-in component props, like `<X title="Hello, World" />`
+const {title} = Astro.props;
+// Fetch external data, even from a private API or database
+const data = await fetch('SOME_SECRET_API_URL/users').then(r => r.json());
+---
+<!-- Your template here! -->
+```
+
+سياج الكود "code fence" مصمّم لضمان أنّ جافاسكربت الذي ستكتبه سيكون "محاط بسياج". لن يعرض في تطبيقك على جانب العميل، ولن يقع بين يدي مستخدميك. يمكنك هنا كتابة الكود بأمان، حتى إذا كانت تكلفة التنفيذ عالية أو كان يتضمن معلومات حسّاسة (مثل: إستدعاء قاعدة البيانات الخاصة بك) دون القلق من أن ينتهي بها المطاف في متصفح مستخدمك.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:53]: محاطًا بسياج.

overlooked by me in the initial review 😅

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+```
+
+## المداخل
+
+`<slot />` هو عنصر نائب لمحتوى HTML الخارجي، مما يسمح لك بحقن (أو "ادخال") عناصر من ملفّات أخرى إلى قالب مكوّنك.
+
+بشكل افتراضي، سيتم تصيير كل العناصر الداخلية الممرّرة لمكوّن في `<slot />` الخاص به.
+
+:::note
+بعكس _الخاصيات _، التي هي سمات "attributes" يتم تمريرها إلى مكوّن أسترو والمتاحة للإستخدام في المكوّن الخاص بك باستخدام `Astro.props`، _المداخل "slots"_ تُصيّر عناصر HTML الداخليّة حيث كتبت.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:173]: extra space between one of the underscores and the word inside causes it not to be Italicized

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 21]
```diff
@@ -0,0 +1,298 @@
...
+لاحظ أن المدخلات المسماة  يجب أن تكون تابعة بشكل مباشر للمكوّن. لا يمكنك ان تمرر مدخلات مسماة داخل عناصر متداخلة.
+
+:::tip
+يمكن ايضًا أن تمرر المدخلات المسماة إلى [مكوّنات أطر واجهة المستخدم](/ar/core-concepts/framework-components/)!
+:::
+
+
+### المحتوى الإحتياطي للمداخل
+
+بإمكان المداخل ايضًا ان تصّير **محتوى احتياطي**. عندما لا يكون هنالك عنصر تابع مطابق مُرّرَ إلى المدخل، عنصر `<slot />` سيُصيّر عنصر تابع نائب خاص به.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:259]:
```suggestion
بإمكان المداخل أيضًا تصيّير **محتوى احتياطي**. عندما لا يكون هنالك عنصر تابع مطابق مُرّرَ إلى المدخل، عنصر `<slot />` سيُصيّر عنصر تابع نائب خاص به.
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 22]
```diff
@@ -0,0 +1,298 @@
...
+    <p>This is my fallback content, if there is no child passed into slot</p>
+  </slot>
+  <Footer />
+</div>
+```
+
+
+## مكوّنات HTML 
+
+يدعم أسترو استيراد واستخدام ملفات `.html` كمكوّنات أو وضع هذه الملفات في الدليل الفرعي `src/pages` كصفحات. قد ترغب باستخدام مكوّنات HTML اذا كنت تعيد استخدام الكود من موقع ويب مبني بدون استخدام أي اطار عمل "framework"، أو اذا كنت ترغب بالتأكد من أن مكوّنك لا يحتوي على أي ميزات ديناميكية.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:284]: - missing slash after pages to indicate that it is a folder
- إذا، إطار، جميعها همزات وصل

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 23]
```diff
@@ -0,0 +1,298 @@
...
+
+يجب ات تحتوي مكوّنات HTML على HTML صالح فقط، وبالتالي، تفتقر إلى ميزات أسترو الأساسية:
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:286]:
```suggestion
يجب أن تحتوي مكوّنات HTML على HTML صالح فقط، وبالتالي تفتقر إلى ميزات مكوّن أسترو الأساسية:
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 24]
```diff
@@ -0,0 +1,298 @@
...
+
+- لا يدعمون frontmatter، أو الإستيراد على جانب الخادم (Server-side imports)، أو التعابير الديناميكية.
+- أي وسوم `<script>` تُركت غير مجمّعة, سيتم التعامل معها كما لو انها تحتوي على `is:inline`.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:289]:
```suggestion
- أي وسوم `<script>` تُركت غير مجمّعة، سيتم التعامل معها كما لو انها تحتوي على `is:inline`.
```

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 25]
```diff
@@ -0,0 +1,298 @@
...
+- يمكنهم فقط [إسناد الموارد الموجودة في مجلد `public/`](/ar/guides/images/#public) 
+
+:::note
+[عنصر `<slot />`](/en/core-concepts/astro-components/#slots) الموجود داخل مكوّن HTML سيعمل كما يعمل في مكوّن أسترو. لإستخدام عنصر ["HTML Web Component Slot"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot) بدلًا من ذلك، أضف  `is:inline` إلى عنصر `<slot />` الخاص بك
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:293]: missing a full stop

**Review by g9wz — CHANGES_REQUESTED** [review]
minor last touches ✍🏻

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 26]
```diff
@@ -0,0 +1,298 @@
...
+```
+
+## المداخل
+
+`<slot />` هو عنصر نائب لمحتوى HTML الخارجي، مما يسمح لك بحقن (أو "ادخال") عناصر من ملفّات أخرى إلى قالب مكوّنك.
+
+بشكل افتراضي، سيتم تصيير كل العناصر الداخلية الممرّرة لمكوّن في `<slot />` الخاص به.
+
+:::note
+بعكس _الخاصيات _، التي هي سمات "attributes" يتم تمريرها إلى مكوّن أسترو والمتاحة للإستخدام في المكوّن الخاص بك باستخدام `Astro.props`، _المداخل "slots"_ تُصيّر عناصر HTML الداخليّة حيث كتبت.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:173]: Screenshot 2023-07-29 at 2 08 51 AM

this has not been resolved

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 27]
```diff
@@ -0,0 +1,298 @@
+---
+title: المكوّنات
+description: مقدمة إلى بناء المكوّنات في .astro 
+i18nReady: true
+---
+
+**مكوّنات أسترو** هي اللبنات الأساسية لأي مشروع أسترو. إنها مكوّنات قوالب HTML فقط بلا وقت تشغيل من جانب العميل. يمكنك التعرف على مكوّن ويب أسترو عن طريق إمتداد الملف: `.astro`
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:7]: kindly remove any instance of "ويب أسترو", mentioning web is not needed nor used originally

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 28]
```diff
@@ -0,0 +1,298 @@
...
+يدعم أسترو استيراد واستخدام ملفات `.html` كمكوّنات أو وضع هذه الملفات في الدليل الفرعي `src/pages/` كصفحات. قد ترغب باستخدام مكوّنات HTML اذا كنت تعيد استخدام الكود من موقع ويب مبني بدون استخدام أي اطار عمل "framework"، أو اذا كنت ترغب بالتأكد من أن مكوّنك لا يحتوي على أي ميزات ديناميكية.
+
+يجب أن تحتوي مكوّنات HTML على HTML صالح فقط، وبالتالي تفتقر إلى ميزات مكوّن أسترو الأساسية:
+
+- لا يدعمون frontmatter، أو الإستيراد على جانب الخادم (Server-side imports)، أو التعابير الديناميكية.
+- أي وسوم `<script>` تُركت غير مجمّعة، سيتم التعامل معها كما لو انها تحتوي على `is:inline`.
+- يمكنهم فقط [إسناد الموارد الموجودة في مجلد `public/`](/ar/guides/images/#public) 
+
+:::note
+[عنصر `<slot />`](/en/core-concepts/astro-components/#slots) الموجود داخل مكوّن HTML سيعمل كما يعمل في مكوّن أسترو. لإستخدام عنصر ["HTML Web Component Slot"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot) بدلًا من ذلك، أضف  `is:inline` إلى عنصر `<slot />` الخاص بك.
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:293]: change the first link to navigate locally, it's currently routing to the en page

change "/en/" with "/ar/" && "#slots" with "#المداخل"

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 29]
```diff
@@ -0,0 +1,298 @@
...
+```
+
+### البرنامج النصّي للمكوّن
+
+أسترو يستخدم سياجًا للكود "code fence" (`---`) للتعريف بالبرنامج النصي للمكوّن في ملف أسترو الخاص بك. اذا كنت قد استخدمت MarkDown من قبل، فقد تكون ملمًّا بمفهوم يسمى *"frontmatter"*. فكرة مكوّن البرنامج النصّي في أسترو مستوحاة مباشرة من هذا المفهوم.
+
+يمكنك استخدام البرنامج النصّي للمكوّن لكتابة أي كود جافاسكربت تحتاجه لتصيير قالبك. يمكن أن يشمل هذا:
+
+- استيراد مكوّنات أسترو أخرى
+- استيراد مكوّنات أطر عمل أخرى، مثل رياكت
```
Comment (g9wz) [inline src/content/docs/ar/core-concepts/astro-components.mdx:34]: add "(React)" at the end of the sentance

**Review by g9wz — CHANGES_REQUESTED** [review]
kindly fix the following

**Comment by g9wz** [comment]
sync your fork after that, I want to compare & pr an updated nav.ts file

**Review by g9wz — APPROVED** [review]
@Yan-Thomas LGTM 👍🏻

**Review by g9wz — APPROVED** [review]

File: src/content/docs/ar/core-concepts/astro-components.mdx [hunk 30]
```diff
@@ -0,0 +1,297 @@
...
+
+:::
+
+### قالب المكوّن
+
+قالب المكوّن موجود أسفل سياج الكود "code fence" ويحدد مخرجات HTML الخاصة بمكوّنك.
+
+اذا قمت بكتابة HTML العادي هنا، فسيقوم المكوّن الخاص بك بتصيير ذلك الـ HTML في أي صفحة أسترو يتم استيرادها واستخدامها.
+
+ومع ذلك، يدعم [ بناء قالب المكوّن لأسترو](/ar/core-concepts/astro-syntax/) كذلك **تعابير جافاسكربت**، وسوم [`<style>`](/ar/guides/styling/#styling-in-astro) و[`<script>`](/guides/client-side-scripts/#using-script-in-astro) لأسترو والمكوّنات المستوردة و[توجيهات أسترو الخاصة](/reference/directives-reference/). البيانات والقيم المعرّفة في البرنامج النصي للمكوّن يمكن إستخدامها في قالب المكوّن لإنتاج HTML  مُنشأ ديناميكيًا.
```
Comment (Yan-Thomas) [inline src/content/docs/ar/core-concepts/astro-components.mdx:66]:
```suggestion
ومع ذلك، يدعم [ بناء قالب المكوّن لأسترو](/ar/core-concepts/astro-syntax/) كذلك **تعابير جافاسكربت**، وسوم [`<style>`](/ar/guides/styling/#styling-in-astro) و[`<script>`](/ar/guides/client-side-scripts/#using-script-in-astro) لأسترو والمكوّنات المستوردة و[توجيهات أسترو الخاصة](/ar/reference/directives-reference/). البيانات والقيم المعرّفة في البرنامج النصي للمكوّن يمكن إستخدامها في قالب المكوّن لإنتاج HTML مُنشأ ديناميكيًا.
```

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks @Somespi 🙌

**Pull request merged**